### PR TITLE
reduce number of test legs we have to match a pairwise test approach

### PR DESCRIPTION
**Pull request by jmarolf**
using the pict model from [here](https://gist.github.com/jmarolf/bd76ade8201866cdb7b0adbb58b1a4a3) we should be able to reduce the number of tests we are running while getting the same coverage.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -30,15 +30,6 @@ jobs:
     configuration: Release
     queueName: Build.Windows.Amd64.VS2022.Pre.Open
 
-- template: eng/pipelines/test-windows-job.yml
```
Comment (RikkiGibson) [inline azure-pipelines.yml:33]: I was surprised to see these jobs removed entirely. I thought we could adjust the configuration to ensure they only run on rolling builds and not on PRs. Was I wrong on that?

Comment (jmarolf) [reply azure-pipelines.yml:33]: I do not think that is possible. @JoeRobich is looking into some stuff but are far as I can tell that is well beyond what azdo supports. Furthermore, I see no reason to keep these around. Someone needs to come to me with a real case where a bug only reproduces if all 3 variables in our test matrix are just so.

Comment (RikkiGibson) [reply azure-pipelines.yml:33]: Simply include the template only if `eq($(Build.Reason), "PullRequest")`. https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml

File: azure-pipelines.yml [hunk 2]
```diff
@@ -77,21 +79,22 @@ jobs:
...
-
-- template: eng/pipelines/test-windows-job.yml
-  parameters:
-    testRunName: 'Test Windows Desktop Release 64'
-    jobName: Test_Windows_Desktop_Release_64
-    buildJobName: Build_Windows_Release
-    testArtifactName: Transport_Artifacts_Windows_Release
-    configuration: Release
-    testArguments: -testDesktop -test64
+    testArguments: -testDesktop -test64 -helixQueueName Windows.10.Amd64.Server19H1.ES.Open
```
Comment (RikkiGibson) [inline azure-pipelines.yml:87]: I imagine this is to complement the fact that we don't run `Test Windows Desktop Release 64` on PRs?

**Review by RikkiGibson — APPROVED** [review]
LGTM, I would just ask that we also try doing a manual branch build of [roslyn-ci](https://dev.azure.com/dnceng/public/_build?definitionId=15) to demonstrate that the extra jobs run on the rolling build. Will probably be necessary to push the branch to dotnet/roslyn.

**Comment by jmarolf** [comment]
@RikkiGibson queued CI build [here](https://dev.azure.com/dnceng/public/_build/results?buildId=1461621). Appears to have the additional runs that we want

**Pull request merged**